astronomer/providers/amazon/aws/hooks/s3.py

```
@@ -13,7 +22,201 @@ class S3HookAsync(AwsBaseHookAsync):
+            )
+            self.log.debug("New objects: %s", current_objects - previous_objects)
+            last_activity_time = datetime.now()
+            inactivity_seconds = 0
+            previous_objects = current_objects
+            return False
+
+        if len(previous_objects) - len(current_objects):
+            # During the last poke interval objects were deleted.
+            if allow_delete:
+                deleted_objects = previous_objects - current_objects
+                previous_objects = current_objects
+                last_activity_time = datetime.now()
+                self.log.info(
+                    "Objects were deleted during the last poke interval. Updating the "
+                    "file counter and resetting last_activity_time:\n%s",
+                    deleted_objects,
+                )
+                return False
+
+            raise AirflowException(
+                f"Illegal behavior: objects were deleted in"
+                f" {os.path.join(bucket_name, prefix)} between pokes."
+            )
+
+        if last_activity_time:
+            inactivity_seconds = int((datetime.now() - last_activity_time).total_seconds())
+        else:
+            # Handles the first poke where last inactivity time is None.
+            last_activity_time = datetime.now()
+            inactivity_seconds = 0
+
+        if inactivity_seconds >= inactivity_period:
+            path = os.path.join(bucket_name, prefix)
+
+            if current_num_objects >= min_objects:
+                self.log.info(
+                    "SUCCESS: \nSensor found %s objects at %s.\n"
+                    "Waited at least %s seconds, with no new objects uploaded.",
+                    current_num_objects,
+                    path,
+                    inactivity_period,
+                )
+                return True
+
+            self.log.error("FAILURE: Inactivity Period passed, not enough objects found in %s", path)
```

After inactivity period if no objects are found we need to raise an exception back to the  dag ??

---

[Done](https://github.com/astronomer/astronomer-providers/pull/86/commits/d0c479f4de534c1ba157b4d3fded4de7bb22c4ad)